### ISSUE TITLE
[Multiscan] NoMethodError: Undefined method 'zip' for nil:nilClass when running parallel tests

### ISSUE THREAD

**Issue by szymon-kazmierczak** [lyndsey-ferguson/fastlane-plugin-test_center#152]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description
test_center 3.8.4

Running tests in parallel with 3 simulators, on xcode 11/ios13, I'm seeing the following exception at the end of the test run:

```
bundler: failed to load command: fastlane (/Users/mobile/.rbenv/versions/2.5.1/bin/fastlane)
NoMethodError: [!] undefined method `zip' for nil:NilClass
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:97:in `collate_infoplist'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:25:in `block in collate_bundles'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:23:in `foreach'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:23:in `collate_bundles'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:14:in `block in run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:13:in `each'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/collate_test_result_bundles.rb:13:in `run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/report_collator.rb:127:in `collate_test_result_bundles'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/report_collator.rb:24:in `collate'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:204:in `collate_batched_reports_for_testable'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:165:in `block in collate_batched_reports'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:164:in `each'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:164:in `collate_batched_reports'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:146:in `run_test_batches'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/helper/multi_scan_manager/runner.rb:53:in `run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.8.4/lib/fastlane/plugin/test_center/actions/multi_scan.rb:23:in `run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:256:in `block (2 levels) in execute_action'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:235:in `block in execute_action'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/fast_file.rb:154:in `method_missing'
  Fastfile:66:in `block (2 levels) in parsing_binding'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/lane_manager.rb:56:in `cruise_lane'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
  /Users/mobile/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.131.0/bin/fastlane:23:in `<top (required)>'
  /Users/mobile/.rbenv/versions/2.5.1/bin/fastlane:23:in `load'
  /Users/mobile/.rbenv/versions/2.5.1/bin/fastlane:23:in `<top (required)>'
[11:46:19]: Deleting iPhone-SE-Automation-iOS13 Clone 1 TestCenter::Helper::MultiScanManager::SimulatorHelper<70205828374740>
[11:46:39]: Deleting iPhone-SE-Automation-iOS13 Clone 2 TestCenter::Helper::MultiScanManager::SimulatorHelper<70205828374740>
[11:46:59]: Deleting iPhone-SE-Automation-iOS13 Clone 3 TestCenter::Helper::MultiScanManager::SimulatorHelper<70205828374740>
```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>[INSERT OUTPUT HERE]</pre>
</details>

### Environment

<details>
  <pre>[INSERT OUTPUT HERE]</pre>
</details>

**Comment by szymon-kazmierczak** [comment]
I haven't attached env or full logs as the logs are massive. I noticed the following happening as well, not sure if related or a different issue:

```
[worker 2] ** TEST EXECUTE FAILED **
[worker 2] 
[worker 2] --- xcodebuild: WARNING: Failed to create symlink at /Users/mobile/jenkins/workspace/PROJECT/fastlane/reports/UITESTS-batch-2/UITESTS.test_result.xcresult - a file already exists.
```

on the rerun (due to failed tests). Not sure if its related.

**Comment by lyndsey-ferguson** [comment]
Hm. This is the same as #145, closing this as it is a duplicate.

**Comment by lyndsey-ferguson** [comment]
And, I'm working on it. 😉

**Comment by lyndsey-ferguson** [comment]
Just to double-check, you have result bundles as an option? Can you share how you are calling `multi_scan`?

**Comment by szymon-kazmierczak** [comment]
sure, heres the multiscan

    result = multi_scan(
      workspace: XCODE_WORKSPACE_FILEPATH,
      try_count: 3,
      fail_build: false,
      scheme: scheme,
      testrun_completed_block: test_run_block,
      device: deviceName,
      output_directory: "fastlane/reports",
      result_bundle: "TestResults",
      prelaunch_simulator: true,
      quit_simulators: false,
      parallel_testrun_count: 3
    )

**Comment by lyndsey-ferguson** [comment]
Yup, same problem.
